### ISSUE TITLE
not able to change canonical name label

### ISSUE THREAD

**Issue by henrysecond1** [knative-sandbox/net-istio#973]
Some apps like kiali use `service.istio.io/canonical-name`, `canonical-revision` labels, so we want to manually set the labels.
However, we were not able to set them because the net-istio webhook forcefully set these labels.

- https://github.com/knative-sandbox/net-istio/blob/release-1.7/pkg/defaults/deployment_defaults.go#L69-L80

Can we set labels only if the users do not specify them?

**Comment by nak3** [comment]
> Can we set labels only if the users do not specify them?

Yes, I think it is no problem to change the webhook as you said. What do you think @AngeloDanducci

**Comment by AngeloDanducci** [comment]
I'm always in favor of more freedom of configuration as long as the defaults are good 👍
